Allow userdomain named filetrans to pkcs_slotd_tmpfs_t domain and removing pkcs_tmpfs_filetrans interface and edit pkcs policy files

Koncpa opened the pull request:
**Commit1:** Allow login_userdomain named filetrans to pkcs_slotd_tmpfs_t domain

User can initialized token. But token which is
being initialized by user get a default context user_tmp_t.
Tokens need to have pkcs_slotd_tmpfs_t domain for proper
functionality with services like certmonger and etc.
Need to create named filetrans, due to SELinux conflicts
without specifying object.
Allow named filetrans from login_userdomain to pkcs_slotd_tmpfs_t.

List of all named tokens:https://github.com/opencryptoki/opencryptoki/blob/master/doc/system_resources
Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=2001599

**Commit 2:** Removing pkcs_tmpfs_filetrans interface and edit pkcs policy files

Removing pkcs_tmpfs_filetrans() interface, which is
replaced by pkcs_tmpfs_named_filetrans and add
additional rules to named interface like manage
pkcs_slotd_tmpfs_t files. Fix typo in pkcs.fc.

Koncpa committed:
Allow login_userdomain named filetrans to pkcs_slotd_tmpfs_t domain

User can initialized token. But token which is
being initialized by user get a default context user_tmp_t.
Tokens need to have pkcs_slotd_tmpfs_t domain for proper
functionality with services like certmonger and etc.
Need to create named filetrans, due to SELinux conflicts
without specifying object.
Allow named filetrans from login_userdomain to pkcs_slotd_tmpfs_t.

List of all named tokens:https://github.com/opencryptoki/opencryptoki/blob/master/doc/system_resources
Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=2001599

zpytela commented:
Patriku,

is unconfined_t the only domain which can initialize the tokens?

Koncpa commented:
No Zdeňku,

tokens can be initialize also by service like ipsec or certmonger, which have different labels.

zpytela commented:
How about other usertypes?

Koncpa commented:
You have right, thank you.
DONE!

zpytela commented:
This is what I had in mind. Thinking about it further though, on one hand I wonder if login_userdomain is just not enough:
```
<    attribute login_userdomain;
---
>    attribute userdomain;
>       auditadm_t
>       dbadm_t
>       logadm_t
>       secadm_t
>       staff_wine_t
>       user_wine_t
>       webadm_t
```

And on the other hand  if the existing rules need attention:
```
# sesearch -T -t tmpfs_t -c dir | grep pkcs
type_transition certmonger_t tmpfs_t:dir pkcs_slotd_tmpfs_t;
type_transition ipsec_t tmpfs_t:dir pkcs_slotd_tmpfs_t;
type_transition pkcs_slotd_t tmpfs_t:dir pkcs_slotd_tmpfs_t;
```
and some unification with this commit.

Koncpa committed:
Removing pkcs_tmpfs_filetrans interface and edit pkcs policy files

Removing pkcs_tmpfs_filetrans() interface, which is
replaced by pkcs_tmpfs_named_filetrans and add
additional rules to named interface like manage
pkcs_slotd_tmpfs_t files. Fix typo in pkcs.fc.

Koncpa commented:
Tested new changes, where are removed interfaces and other modifications,  on related pkcsslotd test and everything succesfully pass without AVC's.

zpytela commented on policy/modules/contrib/pkcs.if:
```diff
@@ -118,25 +118,27 @@ interface(`pkcs_getattr_exec_files',`
+        gen_require(`
+                type pkcs_slotd_tmpfs_t;
+        ')
+        allow $1 pkcs_slotd_tmpfs_t:file map;
```
Please use a blank line as a separator between gen_require and allow.

Koncpa replied:
DONE.

zpytela commented on policy/modules/contrib/pkcs.if:
```diff
@@ -118,25 +118,27 @@ interface(`pkcs_getattr_exec_files',`
-##	</summary>
+##      <summary>
+##      Domain allowed access.
+##      </summary>
```
I'd like to see using tabs for indentation whenever possible.

Koncpa replied:
DONE.

zpytela replied:
Please check again pkcs.if and userdomain.te and look for subsequent spaces in the commits.

zpytela commented:
> Allow logindomain

Last nit: the attribute name is login_userdomain.

zpytela commented:
Merging, thank you.
